### PR TITLE
First version of the tree creation function

### PR DESCRIPTION
**Pull request by SDartayet**
- Minor refactor to the tree data structure (from Box<Option> to Option<Box>) to make accessing it less cumbersome.
- Initial version of the tree creation function done, alongside unit tests for it.

### CODE REVIEW

File: rusty-merkle-tree/Cargo.lock
Comment (mechanix97) [inline rusty-merkle-tree/Cargo.lock:1]: add cargo.lock to git ignore

**Review by mechanix97 — APPROVED** [review]
LGTM
Some improvements: add cargo.lock to gitignore, use cargo fmt before yo do a PR, maybe add a test with an empty tree?

File: rusty-merkle-tree/src/main.rs [hunk 1]
```diff
@@ -1,17 +1,178 @@
...
+        while let Some(node_left) = current_layer_iter.next() {
+
+            //Each node has two child nodes, so I get the next node as well
+            let node_right = current_layer_iter.next().unwrap();
+
+            //Calculate the sum of the hashes
+            let mut hasher = DefaultHasher::new();
+
+            //Values need to be casted to u128 to prevent overflow
+            (node_left.hash.0 as u128 + node_right.hash.0 as u128).hash(&mut hasher);
```
Comment (DiegoCivi) [inline rusty-merkle-tree/src/main.rs:52]: I might be wrong, but i think that if you ever have a really big tree (this causes more levels in the tree, so more concatenations would happen) you could have an overflow even for u128. However i think that this is quite unlikely.

Comment (SDartayet) [reply rusty-merkle-tree/src/main.rs:52]: Don't think that'd happen. The sum of the hash values is always hashed afterwards, so you're never adding more than two u64 values at a time.
